server/resourcemanager/resource_manager.go

```diff
@@ -249,46 +282,27 @@ func (m *manager[T]) upsert(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// if there's no ID given, create the resource
```

Comment:
`upsert` was rewriting code already available in the `create` and `update` methods, so a little refactoring reduced code duplication